src/gtest/test_merkletree.cpp

```diff
@@ -337,7 +344,17 @@ TEST(merkletree, vectors) {
     Array path_tests = read_json(std::string(json_tests::merkle_path, json_tests::merkle_path + sizeof(json_tests::merkle_path)));
 
     test_tree<ZCTestingIncrementalMerkleTree, ZCTestingIncrementalWitness>(root_tests, ser_tests, witness_ser_tests, path_tests);
-    test_tree<OldIncrementalMerkleTree, OldIncrementalMerkleTree>(root_tests, ser_tests, witness_ser_tests, path_tests);
+}
+
+TEST(merkletree, emptyroots) {
+    Array empty_roots = read_json(std::string(json_tests::merkle_roots_empty, json_tests::merkle_roots_empty + sizeof(json_tests::merkle_roots_empty)));
+    Array::iterator root_iterator = empty_roots.begin();
+
+    libzcash::EmptyMerkleRoots<64, libzcash::SHA256Compress> emptyroots;
+
+    for (size_t depth = 0; depth <= INCREMENTAL_MERKLE_TREE_DEPTH; depth++) {
+        expect_test_vector(root_iterator, emptyroots.empty_root(depth));
+    }
```

Comment:
The loop guard should be 64 as well. Also, add a static assertion that `INCREMENTAL_MERKLE_TREE_DEPTH <= 64`.